src/ecs/resources/input.rs

```diff
@@ -51,22 +89,33 @@ impl InputHandler {
                             // second `Pressed` event.
                         }
                         Entry::Vacant(entry) => {
-                            entry.insert(KeyQueryState::Queried);
+                            entry.insert(KeyQueryState::NotQueried);
                         }
                     }
                 }
                 Event::KeyboardInput(ElementState::Released, _, Some(key_code)) => {
                     self.pressed_keys.remove(&key_code);
                 }
-                Event::Focused(false) => self.pressed_keys.clear(),
+                Event::MouseInput(ElementState::Pressed, button) => {
+                    if let Entry::Vacant(entry) = self.pressed_mouse_buttons.entry(button) {
```

Comment:
Why is this an `if let` but L84-94 is not. I would consider changing the match into an `if let`. When I did this I didn't think much of `if let`s.